components/scream/src/physics/p3/p3_functions_f90.cpp

```diff
@@ -3235,13 +3241,25 @@ Int p3_main_f(
   P3F::P3HistoryOnly history_only{liq_ice_exchange_d, vap_liq_exchange_d,
                                   vap_ice_exchange_d};
 
+  // load tables
+  view_1d_table mu_r_table_vals;
+  view_2d_table vn_table_vals, vm_table_vals, revap_table_vals;
+  view_ice_table ice_table_vals;
+  view_collect_table collect_table_vals;
+  view_dnu_table dnu;
+  P3F::init_kokkos_ice_lookup_tables(ice_table_vals, collect_table_vals);
+  P3F::init_kokkos_tables(vn_table_vals, vm_table_vals, revap_table_vals, mu_r_table_vals, dnu);
+
+  P3F::P3LookupTables lookup_tables{mu_r_table_vals, vn_table_vals, vm_table_vals, revap_table_vals,
+                                    ice_table_vals, collect_table_vals, dnu};
+
```

Comment:
This was originally in `p3_main()`, now we move it out and hand `p3_main()` another input structure `P3LookupTables`